### PR TITLE
mTLS analyzer ignores non-TCP ports

### PR DESCRIPTION
**Pull request by selmanj**
Relates to #19360

### CODE REVIEW

File: galley/pkg/config/analysis/analyzers/auth/mtls.go [hunk 1]
```diff
@@ -133,6 +133,11 @@ func (s *MTLSAnalyzer) Analyze(c analysis.Context) {
 		svcSelector := k8s_labels.SelectorFromSet(svc.Selector)
 		fqdn := util.ConvertHostToFQDN(svcNs, svcName)
 		for _, port := range svc.Ports {
+			// Ignore UDP for now. Can be included once
+			// https://github.com/istio/istio/issues/1430 is closed.
+			if port.Protocol != "TCP" && port.Protocol != "" {
```
Comment (sushicw) [inline galley/pkg/config/analysis/analyzers/auth/mtls.go]: If we're specifically ignoring UDP, shouldn't the logic match that? It looks like there are several protocols besides TCP and UDP that could be defined here. 

If the intent is actually "ignore everything except TCP" then we should update the comment and PR title/description accordingly.

Comment (selmanj) [reply galley/pkg/config/analysis/analyzers/auth/mtls.go]: There's only one additional protocol: 'SCTP'. I believe it's still an alpha-feature. It felt weird to call that out here.

That said, I will make the comment/PR more explicit that they're ignoring UDP and others.

Comment (selmanj) [reply galley/pkg/config/analysis/analyzers/auth/mtls.go]: Done.

**Pull request merged**